source/Attribute.cpp

```diff
@@ -0,0 +1,424 @@
+/* Attribute.cpp
+Copyright (c) 2023 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Attribute.h"
+
+#include <algorithm>
+#include <limits>
+#include <unordered_map>
+
+using namespace std;
+
+const string Attribute::effectNames[] = {"shields", "hull", "thrust", "reverse thrust", "turn",
+		"cooling", "active cooling", "cloak", "force", "energy", "fuel", "heat", "discharge", "corrosion", "leak", "burn",
+		"ion", "scramble", "slowing", "disruption", "disabled", "minable", "ramscoop", "piercing", "delay", "depleted delay"};
+
+const string Attribute::categoryNames[] = {"shield generation", "hull repair rate",
+		GetEffectName(THRUST), GetEffectName(REVERSE_THRUST), GetEffectName(TURN), GetEffectName(COOLING),
```

Comment:
I'm confused why some are strings and others are method calls
Can you add a comment?

Reply:
Method calls are used when I want to make sure the category and effect names are the same. Strings are used when they are independent.